core/src/components/datetime/datetime.tsx

```diff
@@ -912,6 +867,29 @@ export class Datetime implements ComponentInterface {
         const monthBox = month.getBoundingClientRect();
         if (Math.abs(monthBox.x - box.x) > 2) return;
 
+        /**
+         * If we're force-rendering a month, and we've scrolled to
+         * that month, return that.
+         *
+         * Checking that we've actually scrolled to the forced month
+         * is mostly for safety; in theory, if there's a forced month,
+         * that means a new value was manually set, so we should have
+         * automatically animated directly to it.
+         */
+        const { forceRenderDate } = this;
+        const firstDayEl = month.querySelector('.calendar-day');
+        const dataMonth = firstDayEl?.getAttribute('data-month');
+        const dataYear = firstDayEl?.getAttribute('data-year');
+        if (
+          forceRenderDate !== null &&
+          dataMonth &&
+          dataYear &&
```

Comment:
If the year picker is open while the date changes, selecting a new time brings you back to the month you were last on:


https://github.com/ionic-team/ionic-framework/assets/2721089/c5ef0e66-fbd1-4f69-aa27-e361d62868e4

Reply:
Fixed by https://github.com/ionic-team/ionic-framework/pull/27806/commits/7a3f92751372589c94d3cab63c915c64c4d60c77.